app/logic/Calendar/OWA/OWAEvent.ts

```diff
@@ -133,14 +133,27 @@ export class OWAEvent extends Event {
     */
   }
 
-  async saveCalendarItem() {
-    let request: any = this.itemID ?
+  getItemRequest() {
+    return this.itemID ?
       new OWAUpdateItemRequest(this.itemID, {SendCalendarInvitationsOrCancellations: "SendToAllAndSaveCopy"}) :
       this.parentEvent ?
       new OWAUpdateOccurrenceRequest(this, {SendCalendarInvitationsOrCancellations: "SendToAllAndSaveCopy"}) :
       new OWACreateItemRequest({SendMeetingInvitations: "SendToAllAndSaveCopy"});
+  }
+
+  getOffice365Request() {
+    return this.itemID ?
+      new OWAUpdateOffice365EventRequest(this.itemID) :
+      this.parentEvent ?
+      new OWAUpdateOffice365OccurrenceRequest(this) :
+      // Office 365 requires an explicit saved item folder id
+      new OWACreateOffice365EventRequest({ SavedItemFolderId: { __type: "TargetFolderId:#Exchange", BaseFolderId: { __type: "DistinguishedFolderId:#Exchange", Id: "calendar" } } });
+  }
+
+  async saveCalendarItem() {
+    let request = this.calendar.account.isOffice365() ? this.getOffice365Request() : this.getItemRequest();
     request.addField("CalendarItem", "Subject", this.title, "item:Subject");
-    request.addField("CalendarItem", "Body", this.descriptionHTML ? { __type: "BodyContentType:#Exchange", BodyType: "HTML", Value: this.descriptionHTML } : this.descriptionText ? { __type: "BodyContentType:#Exchange", BodyType: "Text", Value: this.descriptionText } : null, "item:Body");
+    request.addField("CalendarItem", "Body", this.descriptionHTML ? { __type: "BodyContentType:#Exchange", BodyType: "HTML", Value: this.descriptionHTML } : { __type: "BodyContentType:#Exchange", BodyType: "Text", Value: this.descriptionText || "" }, "item:Body");
```

Comment:
Defaults should use `??` instead of `||`, unless you want to specifically include `""` (which is not the case here, because `""` itself is the fallback).

Reply:
(#449 wants to change this anyway...)